lib/bgs/dependents.rb

```diff
@@ -0,0 +1,173 @@
+# frozen_string_literal: true
+
+module BGS
+  class Dependents
+    def initialize(proc_id:, payload:, user:)
+      @proc_id = proc_id
+      @payload = payload
+      @dependents = []
+      @dependents_application = @payload['dependents_application']
+      @user = user
+    end
+
+    # rubocop:disable Metrics/PerceivedComplexity
+    # rubocop:disable Metrics/CyclomaticComplexity
```

Comment:
I hesitate to be that person, but, I agree with Rubocop here. The complexity behind each private method used is begging to be wrapped in distinct objects.

I don't think we need to block the merge on this, but for the sake of future maintenance and debugging, I think we need to immediately follow-up with these methods broken out.

Reply:
Thanks for your feedback and suggestion.  I agree with you...and plan on iterating and refactoring this in the future so will take that into consideration.  Thank you for not blocking this merge for now.